### PR TITLE
[DOC] Fixes mis-wording/typo in mentoring docs

### PR DESCRIPTION
**Pull request by wayneadams**
#### What does this implement/fix? Explain your changes.

These changes fix what looks to be mis-wording/typos in mentoring docs. Though, I'm not 100% sure what the intention was, so this PR is just to start the editing process.

#### Does your contribution introduce a new dependency? If yes, which one?

No

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

### CODE REVIEW

**Comment by wayneadams** [comment]
@all-contributors please add @wayneadams for doc

**Comment by baraline** [comment]
Hi, thanks and nice spot ! Tell us when you are done with the editting so we can review.

**Comment by wayneadams** [comment]
@baraline , I'm done editing and the PR is ready for review.

File: docs/mentoring.md [hunk 1]
```diff
@@ -112,10 +112,8 @@ shapelets.
 
 ##### Expected Outcomes
 
-We expect the mentee engage with the aeon community and produce a performance games
-for the
-We Based on the benchmark of the different implementations, we will evaluate the
-performance gains of the new shapelet transform and the success of this project.
+We expect the mentee to engage with the aeon community and produce a more performant
+implmentation for the shapelet transform that gets accepted into the toolkit.
```
Comment (baraline) [inline docs/mentoring.md:116]: Small typo on `implmentation`

**Review by baraline — CHANGES_REQUESTED** [review]
Just a small typo, otherwise looking good and in line with the other projects, thanks again !

File: docs/mentoring.md (hunk 1 above)
Comment (wayneadams) [reply docs/mentoring.md]: Good eye!

**Review by baraline — APPROVED** [review]
LGTM!

**Pull request merged**